types/react-router-redux/index.d.ts

```diff
@@ -77,3 +79,5 @@ export interface LocationChangeAction {
 }
 
 export function routerMiddleware(history: History): Middleware;
+
+export function createMatchSelector<State extends { router: RouterState }, Params>(path: string): (state: State) => match<Params> | null;
```

Comment:
There's no need for the generic. It's not even used in the test. Try this instead:

```ts
export function createMatchSelector(path: string): (state: { router: RouterState }) => match<{}> | null;
```
In actual usage, you'd use a type assertion to cast it to the expected type:
```ts
const matchSelector = createMatchSelector('/:foo');
const matchWithKnownType = matchSelector(store.getState()) as match<type_foo>;
```

See the last example [here](https://github.com/Microsoft/dtslint/blob/master/docs/no-unnecessary-generics.md).

Then you can revert the `tslint.json` change too.

Reply:
updated definition and reverted tslint.json changes